File: JavaScript/49-Group Anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    let hmap = new Map();
    for (let word of strs){
        let sortedS = [...word].sort().join("")
        if (!hmap.has(sortedS)){
            hmap.set(sortedS, []);
        }
        hmap.get(sortedS).push(word);
    }
    
    return [...hmap.values()];
};

File: C++/49-Group Anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> hmap;
        vector<vector<string>> res;
        for (string s:strs) {
            string tmp = s;
            sort(s.begin(), s.end());
            hmap[s].push_back(tmp);
        }
        
        for (const auto &kv:hmap) {
            res.push_back(kv.second);
        }
        
        return res;
    }
};